src/MassTransit/Consumers/Batching/BatchConsumer.cs

```diff
@@ -88,7 +88,12 @@ public Task Add(ConsumeContext<TMessage> context, Activity currentActivity)
 
             var messageId = context.MessageId ?? NewId.NextGuid();
 
-            var batchEntry = new BatchEntry(context, context.SentTime ?? context.ReceiveContext.GetSentTime() ?? DateTime.UtcNow,
+            context.TryGetPayload<ITransportSequenceNumber>(out var payload);            
+            ulong sentTimeAsSequenceFallback() => (ulong)(context.SentTime ?? context.ReceiveContext.GetSentTime() ?? DateTime.UtcNow).Ticks;
+
+            var batchEntry = new BatchEntry(
+                context,
+                payload.SequenceNumber ?? sentTimeAsSequenceFallback(),
```

Comment:
As GetSentTime is expensive, can we defer this to only get called when sequenceNumber is not presetn?

Reply:
this is what the method call was there for.  What am I missing?

Reply:
Oh, I didn't see it was a local method.